packages/components/src/Form/src/Form.tsx

```diff
@@ -0,0 +1,102 @@
+import {
+    ButtonContext,
+    TextFieldContext
+} from "@hopper-ui/components";
+import {
+    useResponsiveValue,
+    useStyledSystem,
+    type StyledComponentProps,
+    type ResponsiveProp
+} from "@hopper-ui/styled-system";
+import clsx from "clsx";
+import { forwardRef, type ForwardedRef, type CSSProperties } from "react";
+import { useContextProps, Form as RACForm, type FormProps as RACFormProps } from "react-aria-components";
+
+import { cssModule, SlotProvider } from "../../utils/index.ts";
+
+import { FormContext } from "./FormContext.ts";
+
+import styles from "./Form.module.css";
+
+export const GlobalFormCssSelector = "hop-Form";
+
+export interface FormProps extends StyledComponentProps<RACFormProps> {
+    /**
+     * Whether the form elements are disabled.
+     */
+    isDisabled?: boolean;
+
+    /**
+     * If `true`, the TextField will take all available width.
+     */
+    isFluid?: ResponsiveProp<boolean>;
+
+    /**
+     * The size of the TextField.
+     * @default "md"
+     */
+    size?: ResponsiveProp<"sm" | "md">;
+}
+
+function Form(props: FormProps, ref: ForwardedRef<HTMLFormElement>) {
+    [props, ref] = useContextProps(props, ref, FormContext);
+    const { stylingProps, ...ownProps } = useStyledSystem(props);
+    const {
+        className,
+        children,
+        style,
+        isFluid,
+        isDisabled,
+        size: sizeProp,
+        ...otherProps
+    } = ownProps;
+
+    const size = useResponsiveValue(sizeProp) ?? "md";
+
+    const classNames = clsx(
+        GlobalFormCssSelector,
+        cssModule(
+            styles,
+            "hop-Form",
+            size
```

Comment:
The size isn't actually used in the form css so we probably wouldn't need it here.